File: C++/프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> map;
    
    for(auto& cloth : clothes) {
        if(map.find(cloth[1]) == map.end()) {
            map[cloth[1]] = 1;
        }
        map[cloth[1]]++;
    }
    
    int answer = 1;
    for(const auto& pair : map) {
        answer *= pair.second;
    }
    
    return answer - 1;
}

File: Python3/프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    closet = dict()
    
    for name, cat in clothes:
        if cat not in closet:
            closet[cat] = 1
        closet[cat] += 1
    
    answer = 1
    for key in closet:
        answer *= closet[key]
    
    return answer - 1